docs/developer/guide/writing-a-new-plugin.md

```diff
@@ -5,7 +5,20 @@ Add an entry to the (configuration file)[https://github.com/Azure/autorest/blob/
 add the entry in the corresponding subsection (eg. validators should go in the ```Validators``` section)
 Please note that the ```transform``` step is pre-cursor to most plugins since it transforms the raw json into a usable model. 
```

Comment:
that's not what `transform` does. 99% of the time it does nothing, it handles overrides requested by
- `directive` clauses in the configuration (say someone wants a different method name when generating for Ruby than when generating for C# - no way to express that in Swagger)
- description overrides delivered by our localization engine